modules/utxo-lib/src/bitgo/outputScripts.ts

```diff
@@ -200,19 +207,21 @@ function getRedeemIndex(keyCombinations: [Buffer, Buffer][], signer: Buffer, cos
   throw new Error(`could not find singer/cosigner combination`);
 }
 
-export function createPaymentP2tr(
+function createPaymentP2trCommon(
   pubkeys: Triple<Buffer>,
-  redeemIndex?: number | { signer: Buffer; cosigner: Buffer }
+  redeemIndex?: number | { signer: Buffer; cosigner: Buffer },
+  isMusig2 = false
 ): bitcoinjs.Payment {
   const keyCombinations2of2 = getTaptreeKeyCombinations(pubkeys);
+  const scriptPathKeyCombinations2of2 = isMusig2 ? keyCombinations2of2.slice(1) : keyCombinations2of2;
   if (typeof redeemIndex === 'object') {
-    redeemIndex = getRedeemIndex(keyCombinations2of2, redeemIndex.signer, redeemIndex.cosigner);
+    redeemIndex = getRedeemIndex(scriptPathKeyCombinations2of2, redeemIndex.signer, redeemIndex.cosigner);
   }
-  const redeems = keyCombinations2of2.map((pubkeys, index) =>
+  const redeems = scriptPathKeyCombinations2of2.map((pubkeys, index) =>
     p2trPayments.p2tr_ns(
       {
         pubkeys,
-        depth: index === 0 ? 1 : 2,
+        depth: isMusig2 || index === 0 ? 1 : 2,
       },
       { eccLib }
     )
```

Comment:
L232: we can use `[user, bitgo]` explicitly there

Reply:
Done